### PR TITLE
feat(facebookLogin)

### PR DESCRIPTION
**Pull request by Rinaldo**
Facebook login working. Does not currently support updating info.

### CODE REVIEW

File: auth/facebook.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+const { User } = require('../db/models')
+module.exports = router
+
+if (!process.env.FACEBOOK_APP_ID || !process.env.FACEBOOK_APP_SECRET) {
+
+  console.log('Facebook app ID / secret not found. Skipping Facebook OAuth.')
+
+} else {
+
+  router.get('/', (req, res, next) => {
```
Comment (lordprana) [inline auth/facebook.js]: Should we remove this route?

File: auth/facebook.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    console.log('got to facebook auth')
+    res.send('got to facebook auth')
+  })
+
+  router.post('/', (req, res, next) => {
```
Comment (lordprana) [inline auth/facebook.js]: Should we remove these console.log statements?

File: auth/facebook.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    console.log('incoming token:', req.body.token)
+    axios.get(`https://graph.facebook.com/me?fields=name,id,email,picture&access_token=${req.body.token}`)
```
Comment (lordprana) [inline auth/facebook.js]: We need to handle the case when we pass an invalid access token. You should probably send back a 401 Unauthorized code.

File: db/models/user.js [hunk 4]
```diff
@@ -16,8 +25,26 @@ const User = db.define('user', {
...
             isEmail: true
         }
-    }
+    },
+    facebookId: {
+        type: Sequelize.STRING,
+        unique: true,
 
+    },
+    facebookPic: {
```
Comment (lordprana) [inline db/models/user.js]: Let's change the name to facebookPicUrl

**Review by lordprana — CHANGES_REQUESTED** [review]

**Review by Renriviera — APPROVED** [review]

**Review by lordprana — APPROVED** [review]

**Pull request merged**